pkg/virt-controller/services/multus_annotations.go

```diff
@@ -56,10 +58,13 @@ func (mnap multusNetworkAnnotationPool) toString() (string, error) {
 func generateMultusCNIAnnotation(vmi *v1.VirtualMachineInstance) (string, error) {
 	multusNetworkAnnotationPool := multusNetworkAnnotationPool{}
 
-	multusNonDefaultNetworks := filterMultusNonDefaultNetworks(vmi.Spec.Networks)
-	for i, network := range multusNonDefaultNetworks {
-		multusNetworkAnnotationPool.add(
-			newMultusAnnotationData(vmi, network, fmt.Sprintf("net%d", i+1)))
+	networkNameScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
+	for _, network := range vmi.Spec.Networks {
+		podInterfaceName := networkNameScheme[network.Name]
+		if podInterfaceName != "" && podInterfaceName != namescheme.PrimaryPodInterfaceName {
+			multusNetworkAnnotationPool.add(
+				newMultusAnnotationData(vmi, network, podInterfaceName))
+		}
```

Comment:
I have to say I find the previous code a lot simpler.

Reply:
introduced a function, hope it help readability

Reply:
It's better, but I still think it looks worse - at least, it looked simpler before.

My overall question is since you're trying to unify the interface naming scheme, why do all this ? Couldn't you just have exported a function that names the interfaces (you're doing it), and use that anywhere a pod interface name is expected ? 

What's the advantage of adding this naming scheme type ? afaiu, I can also choose to not use your new type and compute the interface name whatever way I want. 

From my perspective, achieving consistency in the iface naming scheme can be achieved in simpler way, and the real value this PR is adding is increasing the test coverage of the helper function that filter / look for particular interface types from a list of networks.

Reply:
>My overall question is since you're trying to unify the interface naming scheme, why do all this ? Couldn't you just have exported a function that names the interfaces (you're doing it), and use that anywhere a pod interface name is expected ?
What's the advantage of adding this naming scheme type ? afaiu, I can also choose to not use your new type and compute the interface name whatever way I want.

Not sure what you mean by "adding this naming scheme type" - I changed that a few pushes back (as indeed it was an overkill) 
What this PR does now is using a function `CreateNetworkNameScheme` that returns a simple map (eg {"network1": "net1", "network2": "net2"} etc...) - this map is used wherever it is needed, like podnic.go and multus-annotations.go (and later on the SRIOV remapping fix PR).
So I don't understand your concern - can you elaborate?

Reply:
My concern is the code looks more complex now than it did before.

Trying to re-phrase my question; couldn't you just have exported a function that names the interfaces (you're doing it), and use that anywhere a pod interface name is expected ?

What's the advantage of using this approach ?

Anyway, seems you've convinced a maintainer and more people this change is good for the project. I won't fight to overthrow it.

Reply:
>Trying to re-phrase my question; couldn't you just have exported a function that names the interfaces (you're doing it), and use that anywhere a pod interface name is expected ?
What's the advantage of using this approach ?

That sounds similar to what I initially did with the function with the closure input. Or maybe I didn't get to the bottom of what you're saying :).
I think returning a map that can be used multiple times instead of iterating every time over the networks is an advantage. 

>Anyway, seems you've convinced a maintainer and more people this change is good for the project. I won't fight to overthrow it.

ACK, thanks for your input, it was important for me to understand if you have crucial concerns about this.
Seeing you don't - I'll unhold.